client/src/components/HostPage/containers/LeaderboardPage/index.tsx

```diff
@@ -20,15 +40,118 @@ export function LeaderboardPage() {
         if (joined) {
             console.log(joined);
             dispatch({ type: 'question/set', payload: joined }); // Dispatch action to change playerList
-            dispatch({ type: 'gameStats/toggleGameStarted', payload: true}); // Dispatch action to change playerList
-            dispatch({ type: 'gameStats/toggleRoundInProgress', payload: true}); // Dispatch action to change playerList
+            dispatch({ type: 'gameStats/toggleGameStarted', payload: true }); // Dispatch action to change playerList
+            dispatch({ type: 'gameStats/toggleRoundInProgress', payload: true }); // Dispatch action to change playerList
+        }
+    }
+
+    // This function is called when the host starts the next game round
+    const endGame = async () => {
+
+        // Get our socket and tell the server to start a new round
+        const socket: any = socketService.socket;
+        const result = await gameService.endGame(socket).catch((err) => {
+            alert(err);
+        });
+
+        // Update state variables to display the new host screen
+        if (result) {
+            console.log(result);
+            setGameOver(true);
         }
-    } 
+    }
+
+    // Listen for the player join event from roomService and update our state if one joins
+    const handleAttack = () => {
+        if (socketService.socket)
+            gameService.onAttack(socketService.socket, (attacker, target) => {
+
+                const playerHealthObj = Object();
+                playerHealthObj[target] = playerHealth[target] - attackDamage;
```

Comment:
Theoretically this logic goes into the reducer itself, but it's not necessary. I can fix small things like this when cleaning the code later.